app/frontend/src/features/profile/view/leaveReference/LeaveReferencePage.test.tsx

```diff
@@ -0,0 +1,187 @@
+import { leaveReferenceBaseRoute } from "routes";
+import { service } from "service";
+import { getHookWrapperWithClient } from "test/hookWrapper";
+import { getAvailableReferences, getUser } from "test/serviceMockDefaults";
+import { MockedService } from "test/utils";
+
+import LeaveReferencePage from "./LeaveReferencePage";
+
+const getAvailableReferencesMock = service.references
+  .getAvailableReferences as MockedService<
+  typeof service.references.getAvailableReferences
+>;
+const getUserMock = service.user.getUser as MockedService<
+  typeof service.user.getUser
+>;
+
+function renderLeaveFriendReferencePage(referenceType: string, userId: string) {
+  const { wrapper } = getHookWrapperWithClient({
+    initialRouterEntries: [
+      `${leaveReferenceBaseRoute}/${referenceType}/${userId}`,
+    ],
+  });
+
+  render(
+    <Route path={`${leaveReferenceBaseRoute}/:referenceType/:userId`}>
+      <LeaveReferencePage />
+    </Route>,
+    { wrapper }
+  );
+}
+
+function renderLeaveRequestReferencePage(
+  referenceType: string,
+  userId: string,
+  hostRequest: number
+) {
+  const { wrapper } = getHookWrapperWithClient({
+    initialRouterEntries: [
+      `${leaveReferenceBaseRoute}/${referenceType}/${userId}/${hostRequest}`,
+    ],
+  });
+
+  render(
+    <Route
+      path={`${leaveReferenceBaseRoute}/:referenceType/:userId/:hostRequest`}
+    >
+      <LeaveReferencePage />
+    </Route>,
+    { wrapper }
+  );
+}
+
+describe("LeaveReferencePage", () => {
+  beforeEach(() => {
+    getUserMock.mockImplementation(getUser);
+    getAvailableReferencesMock.mockImplementation(getAvailableReferences);
+  });
+
+  describe("When the reference type is invalid", () => {
+    beforeEach(() => {
+      renderLeaveFriendReferencePage("hello", "1");
+    });
+
+    it("Returns an error", async () => {
+      const errorAlert = await screen.findByRole("alert");
+      expect(
+        within(errorAlert).getByText(INVALID_REFERENCE_TYPE)
+      ).toBeVisible();
+    });
+
+    it("does not show the form", () => {
+      expect(
+        screen.queryByRole("heading", {
+          name: "You met with Friendly Cow",
+        })
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  describe("When the reference type is friend", () => {
+    describe("and the users are friends", () => {
+      beforeEach(() => {
+        renderLeaveFriendReferencePage("friend", "5");
+      });
+
+      it("verifies that the review type is available", async () => {
+        expect(getAvailableReferencesMock).toHaveBeenCalledTimes(1);
+        expect(getAvailableReferencesMock).toHaveBeenCalledWith({ userId: 5 });
+      });
+
+      it("does not return an error", () => {
+        expect(screen.queryByRole("alert")).not.toBeInTheDocument();
+      });
+
+      it("displays the form", async () => {
+        expect(
+          await screen.findByRole("heading", {
+            name: "You met with Friendly Cow",
+          })
+        ).toBeInTheDocument();
+      });
```

Comment:
Note for future PR perhaps, but would be good to assert the rest of the form exists too rather than just the header alone as we can't be certain the rest of the form is not missing if there are no assertions on them.